src/common/LoadingBanner.js

```diff
@@ -2,75 +2,56 @@
 
 import React, { PureComponent } from 'react';
 import { View } from 'react-native';
+import Bar from 'react-native-progress/Bar';
 
 import type { Dispatch } from '../types';
 import { connect } from '../react-redux';
 import { getLoading } from '../selectors';
-import { Label, LoadingIndicator } from '.';
 import type { ThemeData } from '../styles';
-import { ThemeContext, createStyleSheet } from '../styles';
-
-const key = 'LoadingBanner';
-
-const styles = createStyleSheet({
-  block: {
-    flexDirection: 'row',
-    justifyContent: 'center',
-    alignItems: 'center',
-    backgroundColor: 'hsl(6, 98%, 57%)',
-  },
-  none: { display: 'none' },
-});
+import { ThemeContext, BRAND_COLOR, NAVBAR_SIZE } from '../styles';
 
 type SelectorProps = $ReadOnly<{|
   loading: boolean,
 |}>;
 
 type Props = $ReadOnly<{|
-  spinnerColor?: 'black' | 'white' | 'default',
-  textColor?: string,
-  backgroundColor?: string,
+  color: string,
+  unfilledColor: string | null,
 
   dispatch: Dispatch,
   ...SelectorProps,
 |}>;
 
 /**
- * Display a notice that the app is connecting to the server, when appropriate.
+ * Display a indeterminate loading-bar while the app is connecting to the server, when appropriate.
  */
 class LoadingBanner extends PureComponent<Props> {
   static contextType = ThemeContext;
   context: ThemeData;
 
+  static defaultProps = {
+    color: BRAND_COLOR,
+    unfilledColor: null,
```

Comment:
nit: can simplify slightly by dropping this default, and saying `unfilledColor?: string` in the type.

After all, we're not really using `null` in particular here :slightly_smiling_face: -- all we do is hand it to `??` which tests if the value is either null or undefined.

Reply:
Before, I implented just as you said, but that time I didn't try night mode, after a number of reviews given by @WesleyAC  , I encountered with night mode but the way loadingBar was implented doesn't suit to night mode that thanks to @WesleyAC , If `unfilledColor` dropped from default then according to flow-type we have to pass unfilled color from every screen where we are using this loading-bar as of now Props is being passed from only one screen that is stream(ChatNav)

according to my logic, we are passing unfilled color props only from stream(ChatNav), if we are passing this unfilled color from every screen or if we don't have the night mode then type String with having some default white color would be good. But here I added null because we cant fetch `this.context.backgroundColor`(we need it when app turns to night mode) and assigned to unfilled color in default parenthesis. That's why I added null, and if any screen passes props then null will be replaced by props and if not then condition ` unfilledColor={unfilledColor ?? this.context.backgroundColor}`  will take the contex backgroundcolor based on nightMode or daymode, so here null is being used to decide `unfilledcolor`

Reply:
> If `unfilledColor` dropped from default then according to flow-type we have to pass unfilled color

Did you try making the prop optional? That's this syntax that I mentioned:
```js
unfilledColor?: string,
```
with the `?` before the `:`. That's how you tell Flow that a prop (or a property of any object) should have a certain type if it's there, but also it's OK if it's not there at all.

I agree we don't want to be having to explicitly pass this in from each caller.